doc/src/tutorial/compile_cmake.dox

```diff
@@ -1,57 +1,251 @@
+
+From an alternative perspective, if you are familiar with autotools, you'll realize that CMake serves a similar purpose to a `./configure` script. Both generate makefiles for the project you're currently building. Therefore, the CMake equivalent of the following commands:
+
+    $ /path/to/portaudio/source/configure --prefix=/install_location # Assume we're in the build directory 'pabuild'
+    $ make
+    $ make install
+
+would be:
+
+    $ cmake -DCMAKE_INSTALL_PREFIX=/install_location /path/to/portaudio/source # Assume we're in the build directory 'pabuild'
+    $ make
+    $ make install
+
+@subsection compile_cmake_audio_backends Audio Backend Developer Packages
+
+Note that on Linux, you will need to install additional audio developer packages, otherwise CMake will report no usable backends. On Debian, Ubuntu, and their derivatives, you can install one or more of the following developer packages using `apt` to get a usable audio backend:
+
+- `libasound2-dev` for ALSA
+- `libjack-dev` for JACK
+
+TODO: Add PulseAudio developer packages.
+
+Notice that OSS is excluded here. Audio devices will not show up under OSS on most modern Linux distributions, which can confuse end-users.
+
+If you get to this point and you have already compiled PortAudio, delete your build folder and start the PortAudio build instructions again once you've installed the developer packages.
+
+@section compile_cmake_examples Building Examples
+
+@note These instructions apply to **ALL** platforms.
+
+To build examples, simply add `-DPA_BUILD_EXAMPLES=ON` to your configure command and rebuild PortAudio as needed. See @ref compile_cmake_general for more information about build options. Examples will be located in your build folder in the `examples/` directory.
+
+@section compile_cmake_tests Building Tests
+
+@note
+    - These instructions apply to **ALL** platforms.
+    - PortAudio does **NOT** support CTest. Contributions are accepted adding support for it.
+
+Building PortAudio with CMake does not automatically build tests. To build tests, simply add `-DPA_BUILD_TESTS=ON` to your configuration command and rebuild PortAudio as needed. See @ref compile_cmake_general for more information about build options. Tests will be located in your build folder in the `test/` directory.
+
+@section compile_cmake_options CMake Build Options
+
+This section contains options that configure your build. Some options are specific to PortAudio while others are general CMake options.
+
+@subsection compile_cmake_options_general General CMake Options
+
+- `--configure <buildtype>`: Changes the build type. This can be `Release` for a release build, `Debug` for a debug build, `RelWithDebInfo` for a release build but with debugging symbols, `MinSizeRel` for the smallest build possible, and `None` for no build type.
+
+- `-DCMAKE_INSTALL_PREFIX=/install_prefix`: When running `make install` or similar, install to `/install_prefix` instead of CMake's default of `/usr`.
```

Comment:
Again, does this need to be specified at the cmake "configure" phase or "build" phase? I guess it's for configure.